mail/consul.tf

```diff
@@ -0,0 +1,39 @@
+# Discover Consul settings
+module "consul" {
+  source       = "../consul"
+  region       = "${var.region}"
+  environment  = "${var.environment}"
+  account      = "${var.account}"
+  service_name = "${var.service_name}"
+}
+
+# Configure our Consul provider, module can't do it for us
+provider "consul" {
+  address    = "${module.consul.address}"
+  scheme     = "${module.consul.scheme}"
+  datacenter = "${module.consul.datacenter}"
+}
+
+# Publish our outputs into Consul for our application to consume
+resource "consul_keys" "config" {
+  key {
+    name   = "smtp_user"
+    path   = "${module.consul.config_prefix}/SMTP/User"
+    value  = "${aws_iam_access_key.email.id}"
+    delete = true
+  }
+
+  key {
+    name   = "smtp_password"
```

Comment:
Could you remove the name = "" part, it's legacy and will trigger warnings for no good reason.